{
  "url": "https://api.github.com/repos/shap/shap/issues/2352",
  "repository_url": "https://api.github.com/repos/shap/shap",
  "labels_url": "https://api.github.com/repos/shap/shap/issues/2352/labels{/name}",
  "comments_url": "https://api.github.com/repos/shap/shap/issues/2352/comments",
  "events_url": "https://api.github.com/repos/shap/shap/issues/2352/events",
  "html_url": "https://github.com/shap/shap/issues/2352",
  "id": 1098225821,
  "node_id": "I_kwDOBHDcK85BdZid",
  "number": 2352,
  "title": "cannot import name 'DenseData' and 'visualize' from 'shap'",
  "user": {
    "login": "rEstela",
    "id": 44876834,
    "node_id": "MDQ6VXNlcjQ0ODc2ODM0",
    "avatar_url": "https://avatars.githubusercontent.com/u/44876834?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/rEstela",
    "html_url": "https://github.com/rEstela",
    "followers_url": "https://api.github.com/users/rEstela/followers",
    "following_url": "https://api.github.com/users/rEstela/following{/other_user}",
    "gists_url": "https://api.github.com/users/rEstela/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/rEstela/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/rEstela/subscriptions",
    "organizations_url": "https://api.github.com/users/rEstela/orgs",
    "repos_url": "https://api.github.com/users/rEstela/repos",
    "events_url": "https://api.github.com/users/rEstela/events{/privacy}",
    "received_events_url": "https://api.github.com/users/rEstela/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 1,
  "created_at": "2022-01-10T18:20:39Z",
  "updated_at": "2022-01-19T04:08:44Z",
  "closed_at": null,
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "When running the example code (https://libraries.io/pypi/shap):\r\n\r\n```\r\nfrom shap import KernelExplainer, DenseData, visualize, initjs\r\nfrom sklearn import datasets,neighbors\r\nfrom numpy import random, arange\r\n\r\n## print the JS visualization code to the notebook\r\ninitjs()\r\n\r\n## train a k-nearest neighbors classifier on a random subset \r\niris = datasets.load_iris()\r\nrandom.seed(2)\r\ninds = arange(len(iris.target))\r\nrandom.shuffle(inds)\r\nknn = neighbors.KNeighborsClassifier()\r\nknn.fit(iris.data, iris.target == 0)\r\n\r\n## use Shap to explain a single prediction\r\nbackground = DenseData(iris.data[inds[:100],:], iris.feature_names) # name the features\r\nexplainer = KernelExplainer(knn.predict, background, nsamples=100)\r\nx = iris.data[inds[102:103],:]\r\nvisualize(explainer.explain(x))\r\n```\r\n\r\n\r\n\r\n\r\nI got the following errors:\r\n\r\nImportError                               Traceback (most recent call last)\r\n~\\AppData\\Local\\Temp/ipykernel_13492/3852576576.py in <module>\r\n----> 1 from shap import DenseData\r\n\r\nImportError: cannot import name 'DenseData' from 'shap' (C:\\ProgramData\\Anaconda3\\lib\\site-packages\\shap\\__init__.py)\r\n\r\nImportError                               Traceback (most recent call last)\r\n~\\AppData\\Local\\Temp/ipykernel_13492/1790835437.py in <module>\r\n----> 1 from shap import visualize\r\n\r\nImportError: cannot import name 'visualize' from 'shap' (C:\\ProgramData\\Anaconda3\\lib\\site-packages\\shap\\__init__.py)",
  "closed_by": null,
  "reactions": {
    "url": "https://api.github.com/repos/shap/shap/issues/2352/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/shap/shap/issues/2352/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
